src/os/bluestore/BlueStore.cc

```diff
@@ -5511,15 +5511,21 @@ void BlueStore::MempoolThread::_update_cache_settings()
 #define dout_prefix *_dout << "bluestore.OmapIteratorImpl(" << this << ") "
 
 BlueStore::OmapIteratorImpl::OmapIteratorImpl(
-  PerfCounters* _logger, CollectionRef c, OnodeRef& o, KeyValueDB::Iterator it)
+  PerfCounters* _logger, CollectionRef c, OnodeRef& o, KeyValueDB::Iterator it, const omap_iter_seek_t& start_from)
   : logger(_logger), c(c), o(o), it(it)
 {
   logger->inc(l_bluestore_omap_iterator_count);
   std::shared_lock l(c->lock);
   if (o->onode.has_omap()) {
     o->get_omap_key(string(), &head);
     o->get_omap_tail(&tail);
-    it->lower_bound(head);
+    string key;
```

Comment:
IMO this changes the default behavior when start_from is empty. Originally we seek to lower_bound(head) but not it would be lower_bound("")

Reply:
And with this approach we're still not eliminating the duplicate lookup for other get_omap_iterator() callers...